pkg/cache/application.go

```diff
@@ -518,6 +518,10 @@ func (app *Application) onReserving() {
 // onReservationStateChange is called when there is an add or a release of a placeholder
 // If we have all the required placeholders progress the application status, otherwise nothing happens
 func (app *Application) onReservationStateChange() {
+	if app.originatingTask != nil {
+		events.GetRecorder().Eventf(app.originatingTask.GetTaskPod().DeepCopy(), nil, v1.EventTypeNormal, "Placeholder Allocated",
+			"Placeholder Allocated", "Application %s placeholder has been allocated.", app.applicationID)
```

Comment:
To enhance supportability, it's best to include the placeholder name. This requires changes in the `UpdateApplicationReservationEvent` type because right now it does not have the taskID. Once it's there, it can be extracted and passed to this method.

Reply:
Update: after our discussion offline, it's a better not to include the taskID so that the K8s event system can use a counter to collapse the messages. But we can still include the task group name in the message, so we'll have a per task group counter.

Reply:
Ok. Even to include task group name, we will need to make changes in many places passes through couple to app state transitions to pass task details. I am not really sure changes worth the benefit. Can we do when the time comes in?

Reply:
If it's too much hassle, I'm ok with skipping it for the time being.

Reply:
BTW, How does message collapse happens? Is it going to be based on action or reason or note? Because you asked me to use "GangScheduling" for action and reason later.

Reply:
The action/reason is usually a short string, a single word. So that's why I used "GangScheduling" here. I assume the collapsing is based on the "message" itself.

![6581be2c3cb45b721d99b5d0_3Hza2qBOHqaGN1ZSninHyGL3NIsLEKeMWmJT5z4N4NC-6RzUN2SNoXT5fd64oTFxlbKNZ2HasfPIYQlYBS_CReWca0yaoD9Zw8xQtIVWIYElNN1Q9cRhbKadAi2xUNU1Yxih2xrPedHg9PlOAH98Wcc](https://github.com/apache/yunikorn-k8shim/assets/22025731/ecca0943-3937-4ffa-ac12-55a94b817fbe)

Reply:
The image above does not contain the "count" field. Example:

https://stackoverflow.com/questions/40132277/can-kubectl-describe-show-timestamp-of-pod-events

```
$ kubectl get events -o custom-columns=FirstSeen:.firstTimestamp,LastSeen:.lastTimestamp,Count:.count,From:.source.component,Type:.type,Reason:.reason,Message:.message \
                     --field-selector involvedObject.kind=Pod,involvedObject.name=my-pod
FirstSeen              LastSeen               Count   From                Type      Reason             Message
2020-01-21T16:30:25Z   2020-01-21T16:30:25Z   1       default-scheduler   Normal    Scheduled          Successfully assigned my-pod to 10.10.1.3
2020-01-21T16:30:26Z   2020-01-21T16:30:26Z   1       kubelet             Normal    Pulling            Pulling image "my-image"
2020-01-21T16:30:26Z   2020-01-21T16:30:26Z   1       kubelet             Normal    Pulled             Successfully pulled image "my-image"
2020-01-21T16:30:26Z   2020-01-21T16:30:26Z   1       kubelet             Normal    Created            Created container my-container
2020-01-21T16:30:27Z   2020-01-21T16:30:27Z   1       kubelet             Normal    Started            Started container my-container
```

Reply:
It seems "reason" would be displayed as is and "action" would be displayed as "message" in events list. I believe collapse should be based on action or message.

Please refer https://dev.to/lucassha/writing-events-to-pod-objects-using-client-go-1p1h.

I think we should use "GangScheduling" as reason and "Placeholder Allocation" as action. We need to follow the pattern in other places too. May need to change in task.go.